src/app/InteractionModelEngine.h

```diff
@@ -87,14 +87,11 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
      * Spec 8.5.1 A publisher SHALL always ensure that every fabric the node is commissioned into can create at least three
      * subscriptions to the publisher and that each subscription SHALL support at least 3 attribute/event paths.
      */
-    static constexpr size_t kMinSupportedSubscriptionsPerFabric           = 2;
-    static constexpr size_t kMinSupportedPathsPerSubscription             = 2;
+    static constexpr size_t kMinSupportedSubscriptionsPerFabric           = 3;
```

Comment:
should constant changing happen in a different PR?

Based on comments this change is correct, however wondering if it should be together with size changes.

Maybe same question of SubscriptionID usage vs uint64

Reply:
Reverted several commits in this PR, and only keep the first few commit that makes the read handler small.